scene/2d/cpu_particles_2d.cpp

```diff
@@ -769,30 +769,30 @@ void CPUParticles2D::_particles_process(double p_delta) {
 			}
 
 			p.seed = seed + uint32_t(i) + i + cycle;
-			uint32_t _seed = p.seed;
+			_rng->set_seed(p.seed);
 
-			p.angle_rand = rand_from_seed(_seed);
-			p.scale_rand = rand_from_seed(_seed);
-			p.hue_rot_rand = rand_from_seed(_seed);
-			p.anim_offset_rand = rand_from_seed(_seed);
+			p.angle_rand = _rng->randf();
+			p.scale_rand = _rng->randf();
+			p.hue_rot_rand = _rng->randf();
+			p.anim_offset_rand = _rng->randf();
 
 			if (color_initial_ramp.is_valid()) {
-				p.start_color_rand = color_initial_ramp->get_color_at_offset(rand_from_seed(_seed));
+				p.start_color_rand = color_initial_ramp->get_color_at_offset(_rng->randf());
 			} else {
 				p.start_color_rand = Color(1, 1, 1, 1);
 			}
 
-			real_t angle1_rad = direction.angle() + Math::deg_to_rad((rand_from_seed(_seed) * 2.0 - 1.0) * spread);
+			real_t angle1_rad = direction.angle() + Math::deg_to_rad((_rng->randf() * 2.0 - 1.0) * spread);
 			Vector2 rot = Vector2(Math::cos(angle1_rad), Math::sin(angle1_rad));
-			p.velocity = rot * Math::lerp(parameters_min[PARAM_INITIAL_LINEAR_VELOCITY], parameters_max[PARAM_INITIAL_LINEAR_VELOCITY], (real_t)rand_from_seed(_seed));
+			p.velocity = rot * Math::lerp(parameters_min[PARAM_INITIAL_LINEAR_VELOCITY], parameters_max[PARAM_INITIAL_LINEAR_VELOCITY], (real_t)_rng->randf());
```

Comment:
```suggestion
			p.velocity = rot * Math::lerp(parameters_min[PARAM_INITIAL_LINEAR_VELOCITY], parameters_max[PARAM_INITIAL_LINEAR_VELOCITY], _rng->randf());
```
`randf()` already returns `real_t`, so not sure what the cast was for.